perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/ResourcesManagerBlImpl.java

```diff
@@ -344,50 +346,55 @@ public boolean isGroupAssigned(PerunSession sess, Group group, Resource resource
 
 	@Override
 	public void assignGroupToResource(PerunSession sess, Group group, Resource resource) throws InternalErrorException, WrongAttributeValueException, WrongReferenceAttributeValueException, GroupAlreadyAssignedException {
-		Vo groupVo = getPerunBl().getGroupsManagerBl().getVo(sess, group);
+		assignGroupsToResource(sess, Collections.singletonList(group), resource);
+	}
 
-		// Check if the group and resource belongs to the same VO
-		if (!groupVo.equals(this.getVo(sess, resource))) {
-			throw new InternalErrorException("Group " + group + " and resource " + resource + " belongs to the different VOs");
-		}
+	@Override
+	public void assignGroupsToResource(PerunSession perunSession, List<Group> groups, Resource resource) throws InternalErrorException, WrongAttributeValueException, WrongReferenceAttributeValueException, GroupAlreadyAssignedException {
+		Set<Member> members = new HashSet<>();
 
-		if(isGroupAssigned(sess, group, resource)) throw new GroupAlreadyAssignedException(group);
+		for(Group g: groups) {
+			Vo groupVo = getPerunBl().getGroupsManagerBl().getVo(perunSession, g);
```

Comment:
I don't think that we need to retrieve whole VO just to check its ID. You can use the following to perform whole check:
```java
getPerunBl().getAttributesManagerBl().checkGroupIsFromTheSameVoLikeResource(perunSession, g, resource);
```
It will introduce thrown exception ``GroupResourceMismatchException``, but its generally expected behavior in our api.

Reply:
Okay, it is rewritten as you mentioned.